_posts/2019-12-18-microservices-frontend.md

```diff
@@ -0,0 +1,49 @@
+![stitching layer](https://lh3.googleusercontent.com/IlvYJ8OsYGdWz9rw_tBeXz7llcpNPjmVlt2_wKNjpFJafeSRQLOpy6-KHiB99hTtSQsMMBzv607Mbjt8krC_JelrC84wqkw7lfHZz19hHVTHCZAmBGCgy1OOExednujpF8jpIFldCNvJUi6vvtFVnYvW1v7-7bK4BjLmYDDEQSkrf8yXcMxtbuQeutNWUynSCnfRhg1H08vhQuVK0Y2miBX82yL53ZCgzhknvb2jk9sSzPbGeNIc6FSRrqIF2R_5J55tpt5imU9L1OxV5iqmUPwa9baVQ0nXOha6lWmlwfzB9OszkV4JHwEZnGxV-udK8cVS2y7aW9KZnhPygTH21eCCOGTMgE2ETrdMFUpqRjKW7KuQ_RVX3CUWUw220AUfkgiR6BDO_hwcUa0304d3FTR_zLLdnTqKVg7lTTvmV_KQIbwXPRmRlAjGue6eYajbi8viReOqiMxH4sDwKbJrngBKlkXOHoGltDxUMxkurXf8mwjYaMpIH4UhbNYNTw2u7a3kuRZoLYAiXT1qpahNRPyeFIzf51DkCrm5AqdvV9dSPPMBas5BQfOGbvDH7hCRlcLTXgC4KbFzjJpbvP_99Xsc8Wzz2RWbu3jWh2N-f-GGWuQtPnXpaug-c1xUFk18-6ELPGEPfzzq6MNRBYDW60XUQ6VR87UMiYy6zdl8MIf4GGH-YHc1nDdE3TInLuBRyv4iE0YrW_Hz3GD0t9EEfWKHuzs_tjHZ8vOF91Y-v5M9tx0=w361-h371-no)
+
+There is also a transparent proxy to make our request go to the right microservice:
+![proxy](https://lh3.googleusercontent.com/3P4IaFMKecfeGqiLfH4AuYfnpI4hqQO6PnhYRmh35XWf7gv4ySxiAM67Zk0u0TmhAcdbduVKZPGU5ZsOPAAJKyOvqYv7s5TWj1azV-jIuegxXniCsb_yKq60Szsebp_qOevRNeBzez3zHo84ujUxUyZuvu_j7hZ7n0sUoTo28ymvsm2BknxUEWlTKMegHfbffTqsvkMqlVksK6-6bxWu_ErI9UJzsFcMFoFvK51yJtMH9gorIlGFFk_K5ZqXo0eWcRnW8MdPvcGzPW9HgoTeouJJI5SGVLOjcrpmcVexUTUzjygSBMlaXyRWkfs7r7VPgiGpyw7iOQGoHa9_cFsQ_B8vggXP-slerQTFdV9c9lkrxbzq0LfRzjwR-49ctyY4G7obVWcFP_CVpxsBfW7jXOcqtDPEgJzdcR912BWnEiiFpvgnKLcl0En5hAVftD0Yj0CP6x7Ubo41T2O8bWCFXuXeSg08fubyLkqgargn_VqzDSHcU9hF4WiguHppFM1lZFc_FiOS79N3oS6PZE4W3vxqnXqrEAQtUPBhW_UORlRVILJGYXYkkB4oZyNs6n-qZp_Pu5qMWR65zgAB7Gx_dWAmnFD7jiYXBUpNAKxdf1qezrtvtzOe-dxKf0aVnfeemYh3-9AcdYx55NCKe1Rm4a4NqUwQPUkTP6vcmNqOXKUhaO7lj11nXAhCJhLR3bmk-Qe3AR5u9k32TPvW3h_twy3gzL1Pj6HG60MMSs-55tic1sw=w449-h301-no)
+
+In production we'll have to setup this routing some other / better way
+
+Second challenge that I bumped into was styling.
+When you use webcomponents they use the shadow dom and css styling is local.
+The way I fixed that now is by using css variables defined in the webcomponents and then set or overridden in the "stitching layer".
+But for a production ready type application you would probably want to make your own [patternfly][2] type project or have some sort of style guide.
+Downside of that is that your introducing more coupling, another way is to have shared css vars, but then you are duplicating components.
+
+Next and final thing that we need to think about is server side rendering.
+Obviously this is something that we would like to have when we have so may different smaller parts and possible different frameworks.
+Server side rendering could potentially also remove the need for a "stitching layer" and when done really well help us with developing these kinds of microservices apps.
+
+One of the conclusions that we can make at this point is, although theoretically we could build all of these frontends in any framework, in practice we would want to minimize the number as it will increase the size of the initial download.
```

Comment:
```suggestion
One of the conclusions that we can make at this point is, although theoretically we could build all of these frontends in any framework, in practice we would want to minimize the number, as it will increase the size of the initial download.
```